src/fdc3/dotnet/DesktopAgent/src/DesktopAgent/Fdc3DesktopAgent.cs

```diff
@@ -73,19 +215,618 @@ public async Task StartAsync(CancellationToken cancellationToken)
+
+                    if (moduleInstance == null) throw ThrowHelper.TargetInstanceUnavailable();
+
+                    var raisedIntentMessageId = StoreRaisedIntentForTarget(messageId, fdc3InstanceId.ToString(), intent, context, sourceFdc3InstanceId);
+
+                    var target = new AppMetadata(
+                        targetAppMetadata.AppId,
+                        fdc3InstanceId.ToString(),
+                        targetAppMetadata.Name,
+                        targetAppMetadata.Version,
+                        targetAppMetadata.Title,
+                        targetAppMetadata.Tooltip,
+                        targetAppMetadata.Description,
+                        targetAppMetadata.Icons,
+                        targetAppMetadata.Screenshots,
+                        targetAppMetadata.ResultType);
+
+                    return RaiseIntentResponse.Success(raisedIntentMessageId, intent, target);
+                }
+                catch (Fdc3DesktopAgentException exception)
+                {
+                    _logger.LogError(exception, "Error while starting module.");
+                    throw;
+                }
+            }
+        }
+    }
+
+    private string StoreRaisedIntentForTarget(
+        int messageId,
+        string targetFdc3InstanceId,
+        string intent,
+        Context context,
+        string sourceFdc3InstanceId)
+    {
+        var invocation = new RaiseIntentResolutionInvocation(
+                                raiseIntentMessageId: messageId,
+                                intent: intent,
+                                originFdc3InstanceId: sourceFdc3InstanceId,
+                                contextToHandle: context);
+
+        _raisedIntentResolutions.AddOrUpdate(
+                        new(targetFdc3InstanceId),
+                        _ =>
+                        {
+                            var resolver = new RaisedIntentResolver();
+                            resolver.AddRaiseIntentToHandle(invocation);
+                            return resolver;
+                        },
+                        (key, oldValue) => oldValue.AddRaiseIntentToHandle(invocation));
+
+        return invocation.RaiseIntentMessageId;
+    }
+
+    //Publishing intent resolution request to the fdc3 clients, they will receive the message and start their intenthandler appropriately, and send a store request back to the backend.
+    private async Task RaiseIntentResolution(string raisedIntentMessageId, string intent, Context context, string targetId, string sourceFdc3InstanceId)
+    {
+        if (_runningApplications.TryGetValue(new(sourceFdc3InstanceId), out var sourceApp))
+        {
+            var sourceAppIdentifier = new AppIdentifier(sourceApp.AppId, sourceFdc3InstanceId);
+            var intentResolutionRequest = MessageBuffer.Factory.CreateJson(new RaiseIntentResolutionRequest(raisedIntentMessageId, context, new ContextMetadata(sourceAppIdentifier)), _jsonSerializerOptions);
```

Comment:
Adding the app id into the metadata seems odd as the message is not originating from the app. Am I missing something?

Reply:
so the whole logic looks like that If an app raises an intent to a literally existing and running app, then the solver app would get the AppIdentifier from the originating one while resolving the intent.
The line above queries the running applications (fdc3 instance ids - FDC3App type dictionary) and sends the ContextMetadata to the target app to solve it with their IntentHandler(context, contextMetadata?).